testsuite/cluster-test/src/experiments/load_test.rs

```diff
@@ -0,0 +1,327 @@
+    pub mempool_handle: Option<(MempoolNetworkSender, MempoolNetworkEvents)>,
+    pub state_sync_handle: Option<(StateSynchronizerSender, StateSynchronizerEvents)>,
+}
+
+impl StubbedNode {
+    async fn launch(node_endpoint: String) -> Self {
+        // generate seed peers config from querying node endpoint
+        let seed_peers = seed_peer_generator::utils::gen_seed_peer_config(node_endpoint);
+
+        // build sparse network runner
+
+        let pfn_config = NodeConfig::default_for_public_full_node();
+
+        // some sanity checks on the network the stubbed node will be running in
+        assert_eq!(
+            pfn_config.full_node_networks.len(),
+            1,
+            "expected only one fn network for PFN"
+        );
+        let network_config = &pfn_config.full_node_networks[0];
+        assert_eq!(network_config.network_id, NetworkId::Public);
+
+        let mut network_builder =
+            NetworkBuilder::create(ChainId::test(), pfn_config.base.role, network_config);
+
+        let state_sync_handle = Some(
+            network_builder
+                .add_protocol_handler(state_synchronizer::network::network_endpoint_config()),
+        );
+
+        let mempool_handle = Some(network_builder.add_protocol_handler(
+            libra_mempool::network::network_endpoint_config(
+                pfn_config.mempool.max_broadcasts_per_peer,
+            ),
+        ));
+        let network_runtime = Builder::new()
+            .thread_name("stubbed-node-network-")
+            .threaded_scheduler()
+            .enable_all()
+            .build()
+            .expect("Failed to start runtime. Won't be able to start networking.");
+
+        network_builder.build(network_runtime.handle().clone());
+
+        network_builder.start();
+
+        // feed the network builder the seed peer config
+        let mut conn_req_tx = network_builder
+            .conn_mgr_reqs_tx()
+            .expect("expecting connectivity mgr to exist after adding protocol handler");
+
+        let new_peer_pubkeys: HashMap<_, _> = seed_peers
+            .iter()
+            .map(|(peer_id, addrs)| {
+                let pubkeys: HashSet<x25519::PublicKey> = addrs
+                    .iter()
+                    .filter_map(NetworkAddress::find_noise_proto)
+                    .collect();
+                (*peer_id, pubkeys)
+            })
+            .collect();
+
+        let conn_reqs = vec![
+            ConnectivityRequest::UpdateAddresses(DiscoverySource::OnChain, seed_peers),
+            ConnectivityRequest::UpdateEligibleNodes(DiscoverySource::OnChain, new_peer_pubkeys),
+        ];
+
+        for update in conn_reqs {
+            conn_req_tx
+                .send(update)
+                .await
+                .expect("failed to send conn req");
+        }
+
+        Self {
+            network_runtime,
+            mempool_handle,
+            state_sync_handle,
+        }
+    }
+}
+
+async fn mempool_load_test(
```

Comment:
I think if we'd better leave load experiment as a framework for load testing. And all these tasks for different components can be grouped into a mod (like health check mod and effects mod)to allow end users have an ability to construct their own load test.

But since now we only have ur commit to mempool and state sync. we can merge this pr and do refactory later when we have more components involve